lib/rdma.pm

```diff
@@ -0,0 +1,78 @@
+# SUSE's openQA tests
+#
+# Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: RDMA(Remote Direct Memory Access) related functions, mainly emulate RDMA
+#
+# To enable RDMA you need a RDMA card or only use emulate function in this lib
+# There are two different emulate technology: rdma_rxe(RDMA over Ethernet emulation), rdma_siw(Software iWARP)
+# Here we use rdma_rxe by default
+#
+# Maintainer: Yong Sun <yosun@suse.com>
+package rdma;
+
+use base Exporter;
+use Exporter;
+use 5.018;
+use strict;
+use warnings;
+use utils;
+use testapi;
+use base 'opensusebasetest';
+use File::Basename;
+use transactional;
+use version_utils qw(is_transactional is_public_cloud is_sle_micro is_sle);
+
+our @EXPORT = qw(
+  install_rdma_dependency
+  modprobe_rdma
+  link_add_rdma
+  link_add_rxe
+  rdma_record_info
+  enable_rdma_in_nfs);
+
+sub install_rdma_dependency {
+    my @deps = qw(
+      rdma-core-devel
+      librdmacm-utils
+      infiniband-diags
+      libibverbs-utils
+    );
+    script_run('zypper --gpg-auto-import-keys ref');
+    if (is_transactional) {
+        trup_install(join(' ', @deps[0 .. $#deps - 1]));
+        reboot_on_changes;
+    }
+    else {
+        zypper_call('in ' . join(' ', @deps));
+    }
```

Comment:
Please have a look on  https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/d4f8719cf7d7450b2bc12d16c300461e8a7a3d95/lib/package_utils.pm#L22

This function unifies package installation on transactional and normal systems (you can avoid above install logic).